src/containers/stats/courttype/sagas/getStatsByCourtType.js

```diff
@@ -0,0 +1,195 @@
+      if (!isDefined(personEKIDCount)) {
+        personEKIDCountMap.set(personEKID, 1);
+        totalParticipantsForCourtType += 1;
+      }
+    });
+    const enrollmentStatusNeighborsByDiversionPlanEKID = fromJS(response.data)
+      .map((neighbors :List) => neighbors
+        .filter((neighbor :Map) => getNeighborESID(neighbor) === enrollmentStatusESID))
+      .map((enrollmentStatusNeighbors :List) => enrollmentStatusNeighbors
+        .map((enrollmentStatusNeighbor :Map) => getNeighborDetails(enrollmentStatusNeighbor)));
+
+    enrollmentStatusNeighborsByDiversionPlanEKID.forEach((enrollmentStatuses :List) => {
+      // fancy sorting for enrollment statuses, in case statuses were entered without dates or out of order:
+      const statusesWithEffectiveDates :List = enrollmentStatuses
+        .filter((status :Map) => isDefined(status.get(EFFECTIVE_DATE)));
+      let sortedEnrollmentStatuses :List = sortEntitiesByDateProperty(statusesWithEffectiveDates, [EFFECTIVE_DATE]);
+      sortedEnrollmentStatuses = sortedEnrollmentStatuses.sort((enrollmentStatus :Map) => {
+        const status = getPropertyValue(enrollmentStatus, [STATUS, 0]);
+        if (status === ENROLLMENT_STATUSES.AWAITING_CHECKIN) return -1;
+        return 0;
+      });
+      const completionStatuses :List = sortedEnrollmentStatuses.filter((statusInList :Map) => {
+        const status = getPropertyValue(statusInList, [STATUS, 0]);
+        return COMPLETION_STATUSES.includes(status);
+      });
+      const mostRecentEnrollmentStatus :Map = completionStatuses.last() || sortedEnrollmentStatuses.last() || Map();
+      const status = getPropertyValue(mostRecentEnrollmentStatus, [STATUS, 0]);
+
+      if (status === ENROLLMENT_STATUSES.COMPLETED || status === ENROLLMENT_STATUSES.SUCCESSFUL) {
+        const currentCount = enrollmentStatusCountsForCourtType.get(ENROLLMENT_STATUSES.COMPLETED);
+        enrollmentStatusCountsForCourtType.set(ENROLLMENT_STATUSES.COMPLETED, currentCount + 1);
+      }
+      if (status === ENROLLMENT_STATUSES.REMOVED_NONCOMPLIANT || status === ENROLLMENT_STATUSES.UNSUCCESSFUL) {
+        const currentCount = enrollmentStatusCountsForCourtType.get(ENROLLMENT_STATUSES.REMOVED_NONCOMPLIANT);
+        enrollmentStatusCountsForCourtType.set(ENROLLMENT_STATUSES.REMOVED_NONCOMPLIANT, currentCount + 1);
+      }
+      if (status === ENROLLMENT_STATUSES.ACTIVE
+        || status === ENROLLMENT_STATUSES.ACTIVE_REOPENED
+        || status === ENROLLMENT_STATUSES.JOB_SEARCH) {
+        const currentCount = enrollmentStatusCountsForCourtType.get(ENROLLMENT_STATUSES.ACTIVE);
+        enrollmentStatusCountsForCourtType.set(ENROLLMENT_STATUSES.ACTIVE, currentCount + 1);
+      }
+      if (status === ENROLLMENT_STATUSES.CLOSED) {
+        const currentCount = enrollmentStatusCountsForCourtType.get(ENROLLMENT_STATUSES.CLOSED);
+        enrollmentStatusCountsForCourtType.set(ENROLLMENT_STATUSES.CLOSED, currentCount + 1);
+      }
```

Comment:
No change required. Using immutable Map.update(...) might make this more readable
 enrollmentStatusCountsForCourtType.update(ENROLLMENT_STATUSES.CLOSED, 0, (count) => count + 1));